src/pocketmine/entity/projectile/Arrow.php

```diff
@@ -74,6 +77,10 @@ public function onUpdate($currentTick){
 
 		if($this->onGround or $this->hadCollision){
 			$this->setCritical(false);
+			if($this->sound === true and $this->level !== null){ //Prevents error of $this->level returning null
```

Comment:
> Prevents error of $this->level returning null 

maybe so... but I'd be more interested to know why such a thing is happening, so it can be fixed properly, as opposed to adding a hack for it.